ultravox/model/ultravox_model.py

```diff
@@ -247,10 +247,16 @@ def prepare_inputs_for_generation(
             **kwargs,
         )
 
-        if is_cache_empty(past_key_values) and audio_values is not None:
-            # We only want to use audio features in the 1st generation step
+        prefill_start_idx = kwargs["cache_position"][0]
```

Comment:
In order to support cached history in multi-turn conversation, past_key_value may not be None for the 1st generation step of a new turn.

Reply:
Are you absolutely sure `cache_positions` may never be None? Is that dependent on the `transformers` version?

Also if you need it, I suggest putting `cache_positions` in the function signature rather than getting it from `kwargs`. That way the type can also be specified.

Reply:
`prepare_inputs_for_generation` is called in `transformers.generation.utils.GenerationMixin`  after `cache_position` is initialized:

`model_kwargs = self._get_initial_cache_position(input_ids, model_kwargs)`

Reply:
Good suggestion about putting `cache_position` in the function signature.